### ISSUE TITLE
[New Model]: RLHFlow/ArmoRM-Llama3-8B-v0.1

### ISSUE THREAD

**Issue by gx-ai-architect** [vllm-project/vllm#11306]
### The model to consider.

https://huggingface.co/RLHFlow/ArmoRM-Llama3-8B-v0.1

### The closest model vllm already supports.

The generative Llama3 model is the closest: 
https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/models/llama.py

### What's your difficulty of supporting the model you want?

It will need to support adding the projection layer to map logits into reward dimensions. 

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.
